bdsx/bds/blockpos.ts

```diff
@@ -103,89 +103,87 @@ export class BlockPos extends NativeStruct {
         }
     }
 
-    relative(facing:Facing, steps:number):BlockPos {
+    relative(facing: Facing, steps: number): BlockPos {
         abstract();
     }
 
     static create(pos: Vec3): BlockPos;
     static create(pos: BlockPos): BlockPos;
-    static create(pos: {x:number, y:number, z:number}): BlockPos;
-    static create(x:number, y:number, z:number):BlockPos;
-    static create(a:number|{x:number, y:number, z:number}, b?:number, c?:number):BlockPos {
-        const v = new BlockPos(true);
-        if(typeof a === "number") {
-            v.x = a;
-            v.y = b!;
-            v.z = c!;
+    static create(pos: { x: number, y: number, z: number }): BlockPos;
+    static create(x: number, y: number, z: number): BlockPos;
+    static create(a: number | { x: number, y: number, z: number }, b: number = 0, c: number = 0): BlockPos {
+        if (typeof a === "number") {
+            const v = new BlockPos(true);
+            v.x = a > 0 ? Math.ceil(a) : Math.floor(a);
+            v.y = b > 0 ? Math.ceil(b) : Math.floor(b);
+            v.z = c > 0 ? Math.ceil(c) : Math.floor(c);
+            return v;
         } else {
-            v.x = a.x;
-            v.y = a.y;
-            v.z = a.z;
+            return BlockPos.create(a.x, a.y, a.z);
         }
-        return v;
     }
 
-    toJSON():VectorXYZ {
-        return {x:this.x, y:this.y, z:this.z};
+    toJSON(): VectorXYZ {
+        return { x: this.x, y: this.y, z: this.z };
     }
 }
 
-BlockPos.prototype.relative = procHacker.js("?relative@BlockPos@@QEBA?AV1@EH@Z", BlockPos, {this:BlockPos, structureReturn:true}, uint8_t, int32_t);
+BlockPos.prototype.relative = procHacker.js("?relative@BlockPos@@QEBA?AV1@EH@Z", BlockPos, { this: BlockPos, structureReturn: true }, uint8_t, int32_t);
```

Comment:
Please don't edit lines to make it your format.
it's hard to check what is your commit.

Reply:
Sorry!